File: eBibliotekaClient/src/app/data/interfaces/Notification.ts
export class Notification {
  id: number;
  text: string;
  title: string;
  recipientID: number;
  senderID: number;

  constructor() {
    this.id = -1;
    this.text = ' ';
    this.title = ' ';
    this.recipientID = -1;
    this.senderID = -1;
  }
}

export class SendNotificationVM {
  text: string;
  title: string;
  recipientID: number;

  constructor() {
    this.text = ' ';
    this.title = ' ';
    this.recipientID = -1;
  }
}
export class NotificationListItemVM {
  id: number;
  text: string;
  title: string;
  date: Date;
  libraryName: string;

  constructor() {
    this.id = -1;
    this.text = ' ';
    this.title = ' ';
    this.date = new Date();
    this.libraryName = '';
  }
}

File: eBibliotekaServer/LibraryModule/Models/Notification.cs
using eBibliotekaServer.AuthModule.Models;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eBibliotekaServer.LibraryModule.Models
{
    public class Notification
    {   [Key]
        public int ID { get; set; }
        public int? SenderID { get; set; }
        [ForeignKey(nameof(SenderID))]
        public Librarian Sender { get; set; }
        public int? RecipientID { get; set; }
        [ForeignKey(nameof(RecipientID))]
        public User Recipient { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
        public DateTime Date { get; set; }
    }
}
